doc/patch-author-guide.md

```diff
@@ -261,58 +224,76 @@ A shadow PID variable is allocated in `do_fork()`: it is associated with the
 current `struct task_struct *p` value, given a string lookup key of "newpid",
 sized accordingly, and allocated as per `GFP_KERNEL` flag rules.
 
-`kpatch_shadow_alloc` returns a pointer to the shadow variable, so we can
+`klp_shadow_get_or_alloc()` returns a pointer to the shadow variable, so we can
 dereference and make assignments as usual.  In this patch chunk, the shadow
 `newpid` is allocated then assigned to a rolling `ctr` counter value:
 ```
+@@ -1797,6 +1798,13 @@ long do_fork(unsigned long clone_flags,
+ 	if (!IS_ERR(p)) {
+ 		struct completion vfork;
+ 		struct pid *pid;
 +		int *newpid;
 +		static int ctr = 0;
 +
-+		newpid = kpatch_shadow_alloc(p, "newpid", sizeof(*newpid),
-+					     GFP_KERNEL);
++		newpid = klp_shadow_get_or_alloc(p, 0, sizeof(*newpid), GFP_KERNEL,
```

Comment:
Maybe have `#define KPATCH_SHADOW_NEWPID 0` somewhere (or enum) and use that instead?  Perhaps along with a comment that the value needs to be globally unique.

Reply:
ok